text/src/autogluon/text/automm/configs/model/fusion_mlp_image_text_tabular.yaml

```diff
@@ -95,6 +101,12 @@ model:
     insert_sep: False
     text_segment_num: 1
     stochastic_chunk: False
+    text_detection_length: 10
+    train_augment_types:
```

Comment:
`train_augment_types` is a general name. We need to try distinguish this key from image transformation keys. How about `text_augment_types` or `text_train_augment_types`? Do we need to emphasize `train` in this key?

Reply:
I follow the style under timm_image. Since train_augment_types is listed under hf_text.

Reply:
`train_augment_types` may also be used in multimodal models, e.g., clip. So, we need to distinguish this key from the image transformation keys.

Reply:
The image augmentation key names also become too general after adding the text augmentation key. I used those keys because only image augmentations were available and there seems no well-known augmentations in other modalities. We can add `image` into the image augmentation keys later. But here we need to add `text` into the text augmentation key.

Reply:
I think only text augmentations are available in hf_text? We can definitely refactor the config later.

Reply:
Both lines 66 and 103 have `train_augment_types`. I guess it's not only under `hf_text`. Anyway, it is easy to make the key name more accurate in this PR. No need to leave the debt to future.

Reply:
Oh, I just got the point. @lzcemma Should we add augmentation to “data”?

Reply:
Moving to `data.text` can avoid the issue, but we would lose the flexibility to customize augmentations for each text backbone if we use multiple text backbones.

Reply:
After a second thought, I agree with @zhiqiangdon on adopting text_augment_types so that it will be consistent.

Reply:
In the AutoMM design, each data processor is tied to one model/backbone. A data processor aims to provide customized input for its model. For example, different text models may need different tokenizers. The model-agnostic data configs are put in the `config.data`, which is used in creating `df_preprocessor`. Based on this design philosophy, everything put in data processors should be customizable for different models. Since augmentation is used in data processor rather than df_preprocessor, it's better we leave it customizable. Hence putting augmentation config under `hf_text` fits the design philosophy.

Reply:
You will have to write it down somewhere. No one will be able to understand without reading an article that explains the design.